### DISCUSSION TITLE
Targets and Quarto projects

### DISCUSSION THREAD

**Discussion by petrbouchal** [ropensci/targets#813] [Ideas]
It would be great if targets could integrate with document/project workflows in the maturing [Quarto framework](https://quarto.org/docs/). Quarto seems to have its own way of [only rendering documents that are out of date (`freeze`)](https://quarto.org/docs/projects/code-execution.html) but presumably these would not pick up changes in upstream targets. 

For projects (an evolution of rmarkdown websites/bookdown projects that I recall were too knotty for targets to handle the project-level rendering), one simple solution that comes to mind would be for {targets} to `touch` quarto files whose upstream targets have changed. (I wonder if this is something that {targets} could provide as a general-purpose functionality for other uses.) Then it could be left to the user to render the whole project in Quarto using the `freeze` functionality (assuming that Quarto detects changes using modification time). 

Separately, it would be great to have dedicated `tar_render`-style functions for the Quarto workflow rendering individual files.

**Comment by wlandau** [comment]
Great set of suggestions! Haven't looked too deeply at Quarto yet, but it is an exciting effort, and I could see a place for `tar_render()`-like target factory.
